p2p/src/network_ext.rs

```diff
@@ -0,0 +1,80 @@
+//! SPDX-License-Identifier: CC0-1.0
+//!
+//! This module implements an extension trait for [`Network`]
+//! with getter methods for the default P2P port and network [`Magic`] bytes.
+
+use crate::Magic;
+
+use bitcoin::{Network, TestnetVersion};
+
+/// Trait that extends [`Network`] by adding getter methods for the default P2P ports and [`Magic`] bytes.
+pub trait NetworkExt {
+    /// The default P2P port for a given [`Network`].
+    fn p2p_port(&self) -> u16;
+    /// The default [`Magic`] for a given [`Network`].
+    fn network_magic(&self) -> Magic;
+}
+
+impl NetworkExt for Network {
+    fn p2p_port(&self) -> u16 {
```

Comment:
The method should be called `default_p2p_port` to signal that the actual port may be different.